### DISCUSSION TITLE
Predictions on data without a `y`

### DISCUSSION THREAD

**Discussion by anthonysirico** [pyg-team/pytorch_geometric#5882] [Q&A] [answered]
So, for a real world scenario, what is the best possible way of performing predictions?

The way I approached is for my test set is here:
```
predictions = []
y = []
model.eval()
correct = 0
for test in test_loader:
    test = test.to(device)
    out = model(test.x, test.edge_index, batch=test.edge_index.new_zeros(test.x.size(0)))
    pred = out.argmax(dim=1)
    y.append(test.y.item())
    predictions.append(pred.item())
    correct += int((pred == test.y).sum())

pred_df = pd.DataFrame()
pred_df['predictions'] = predictions
pred_df['y_true'] = y
```

What changes should be made for when your prediction set does not have a `y` value? Are there any real world examples on this? Specifically graph classification.

**Comment by rusty1s** [comment, marked as answer]
If you do not have ground-truth labels to compare against, all you can do is return the `predictions`, and use them for your use-case. Let me know if I am misunderstanding your question.

**Reply by anthonysirico** [reply]
My scenario is that I have 50,000 data points, I train on 10,000, then predict 40,000. The 10,000 will have ground truth while the 40,000 will not. What would be a suggested code to predict the 40k?

**Reply by anthonysirico** [reply]
I guess what im asking is what is the best way to make predictions?

This way?
```
predictions = []

model.eval()

for test in test_loader:
    test = test.to(device)
    out = model(test.x, test.edge_index, batch=test.edge_index.new_zeros(test.x.size(0)))
    pred = out.argmax(dim=1)

    predictions.append(pred.item())
```
or this way?
```
def test(loader):
    model.eval()
    predictions = []
    for data in loader:  # Iterate in batches over the training/test dataset.
        data = data.to(device)
        out = model(data.x, data.edge_index, data.batch)
        pred = out.argmax(dim=1)  # Use the class with highest probability.
        predictions.append(pred)
    return predictions
```

**Reply by rusty1s** [reply]
The latter one looks correct to me.
